src/data_structures/hashmap.md

```diff
@@ -0,0 +1,20 @@
+### Hashmaps
+
+With arrays you have to store data in an ordered sequence identified with a indx. `Hashmaps` store data in a (key, pair) sequence that you can access by the index referring to another Object(indx is also an Object)
```

Comment:
This misses a pretty needed step of explaining how `.equals` and `.hashCode` work. Without that they would make a `HashMap<Apple, Apple>` and not be primed to understand what is going on.

That or they might have a mutable aggregate as a key.